### ISSUE TITLE
LinkGenerator.GetPathByName() doesn't work

### ISSUE THREAD

**Issue by mohd-akram** [aspnet/AspNetCore#7098]
### Describe the bug
`LinkGenerator.GetPathByName()` doesn't work.

### To Reproduce

1. Create a project with `dotnet new mvc`
2. Change the `HomeController` code to this:

```csharp
    public class HomeController : Controller
    {
        private readonly LinkGenerator _linkGenerator;
        public HomeController(LinkGenerator linkGenerator)
        {
            _linkGenerator = linkGenerator;
        }

        [HttpGet("Test", Name = "Test")]
        public IActionResult Test()
        {
            return Ok(
$@"GetPathByAction: {_linkGenerator.GetPathByAction(HttpContext, "Test")}
GetPathByName: {_linkGenerator.GetPathByName("Test", null)}"
            );
        }
    }
```

3. `dotnet run`
4. Go to `http://localhost:5000/Test`

### Expected output

```
GetPathByAction: /Test
GetPathByName: /Test
```

### Actual output

```
GetPathByAction: /Test
GetPathByName: 
```

### Additional context

```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.103
 Commit:    8edbc2570a

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  10.14
 OS Platform: Darwin
 RID:         osx.10.14-x64
 Base Path:   /usr/local/share/dotnet/sdk/2.2.103/

Host (useful for support):
  Version: 2.2.1
  Commit:  878dd11e62

.NET Core SDKs installed:
  2.2.103 [/usr/local/share/dotnet/sdk]

.NET Core runtimes installed:
  Microsoft.AspNetCore.All 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.All]
  Microsoft.AspNetCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.2.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET Core runtimes or SDKs:
  https://aka.ms/dotnet-download
```

**Comment by AndreaCuneo** [comment]
We're having the same issue, also with Url.Link() equivalent.
We've worked around it by setting EnableEndpointRouting = false; in the MvcOptions.

**Comment by mkArtakMSFT** [comment]
@mohd-akram can you please confirm you're still facing this issue with the most recent 2.2 release?

**Comment by mohd-akram** [comment]
Still a problem with 2.2.4.

**Comment by rynowak** [comment]
This is fixed in 3.0-preview6. You can now use `GetPathByName` and it will understand route names for MVC.

If you're looking for functionality similar to `Url.Route` in 2.2 I would recommend using `LinkGenerator.GetPathByRouteValues`
